spark8t/services.py

```diff
@@ -49,72 +49,71 @@
 class AbstractKubeInterface(WithLogging, metaclass=ABCMeta):
     """Abstract class for implementing Kubernetes Interface."""
 
-    @abstractmethod
-    def with_context(self, context_name: str):
+    defaults: Defaults
+    kube_config_file: Union[None, str, Dict[str, Any]]
+    context_name: Optional[str] = None
```

Comment:
Variable name `context_name` conflicts with a method of the same name. I believe this was supposed to be `_context_name` because that's what is assigned in `__init__` constructor.

Reply:
Oh, yes. This is actually a left over of a refactoring before creating a common init. Strange that the linting did not complain. It is probably allowed since these are static attributes, but I agree, it is confusing. I'll remove them all together.